solver/src/settlement.rs

```diff
@@ -111,26 +117,61 @@ impl Settlement {
         Ok(cursor.into_inner())
     }
 
-    // For now this computes the total surplus of all EOA trades.
-    pub fn objective_value(&self) -> U256 {
-        // TODO: in order to achieve a fair comparison between different solutions, we need to normalize the price
-        // vector as otherwise any price vector scaling by factor x would also scale the objective value by that factor
-        // without affecting the validity of the solution.
-        match self.trades.iter().fold(Some(U256::zero()), |acc, trade| {
-            let sell_token_price = self
-                .clearing_prices
+    fn total_surplus(&self, prices: &HashMap<H160, BigRational>) -> Option<BigRational> {
+        self.trades.iter().fold(Some(num::zero()), |acc, trade| {
+            let sell_token_external_price = prices
                 .get(&trade.order.sell_token)
                 .expect("Solution with trade but without price for sell token");
-            let buy_token_price = self
-                .clearing_prices
+            let buy_token_external_price = prices
                 .get(&trade.order.buy_token)
-                .expect("Solution with trade but without price for sell token");
-            acc?.checked_add(trade.surplus(*sell_token_price, *buy_token_price)?)
-        }) {
+                .expect("Solution with trade but without price for buy token");
+            acc?.checked_add(&trade.surplus(sell_token_external_price, buy_token_external_price)?)
```

Comment:
I thought about that too (btw I haven't found an easy to check that from the code since Ratio<T> is coded for a generic T which doesn't overflow for the specific case of T=BigInt).

However. I believe divisions by 0 are still coerced into None, so we would still need the Option...

Reply:
Actually I'm not sure about what happens for div 0. Will check the source!

Reply:
Found it: https://docs.rs/num-rational/0.1.42/num_rational/struct.Ratio.html#method.checked_div . I guess this means we need to keep the Option, right?

Reply:
I still see a bunch of checked_... that are not checked_div so we could start by removing those and seeing how many options we can remove. For the divs it depends on whether the divisor can really be 0 which I am not sure about from the top of my head.

Reply:
Done here:  119b062